### PR TITLE
Add an option to produce a textfile trace of instructions executed

### PR DESCRIPTION
**Pull request by breqdev**
Useful for debugging.

Adds infrastructure as well in case we want to make the trace more detailed in the future, make other kinds of trace handlers, etc.

### CODE REVIEW

**Comment by breqdev** [comment]
I wanted to avoid a `Box`-heavy API here but I think it's unavoidable -- `impl` doesn't work on `System` since its a trait :(

File: src/trace/file.rs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      file: File::create(filename).expect("Invalid filename"),
+    }
+  }
+}
+
+impl TraceHandler for FileTraceHandler {
+  fn handle(&mut self, trace: &CpuTrace) {
+    self
+      .file
+      .write_all(format!("{:04X}: {:02X}\n", trace.address, trace.opcode).as_bytes())
```
Comment (ava-silver) [inline src/trace/file.rs:23]: could be good to make this buffered with `BufWriter`, i'm happy to implement that

**Review by ava-silver — DISMISSED** [review]
nit: I can implement buffering for the trace but otherwise LGTM!

**Comment by breqdev** [comment]
Rebased on top of main to include changes from #40

**Review by ava-silver — APPROVED** [review]
image

**Comment by breqdev** [comment]
:shipit: :shipit: :shipit:

**Pull request merged**